packages/task/src/browser/quick-open-task.ts

```diff
@@ -721,3 +721,46 @@ export class TaskRunningQuickOpen implements QuickOpenModel {
         });
     }
 }
+
+@injectable()
+export class TaskRestartRunningQuickOpen implements QuickOpenModel {
+
+    @inject(QuickOpenService)
+    protected readonly quickOpenService: QuickOpenService;
+
+    @inject(TaskService)
+    protected readonly taskService: TaskService;
+
+    async onType(_lookFor: string, acceptor: (items: QuickOpenItem[]) => void): Promise<void> {
+        const items: QuickOpenItem[] = [];
+        const runningTasks: TaskInfo[] = await this.taskService.getRunningTasks();
+        if (runningTasks.length <= 0) {
+            items.push(new QuickOpenItem({
+                label: 'No task to restart',
+                run: (): boolean => false,
+            }));
+        } else {
+            runningTasks.forEach((task: TaskInfo) => {
+                items.push(new QuickOpenItem({
+                    label: task.config.label,
```

Comment:
I guess we might also need a group label, as the labels from tasks in a multi root workspace could possibly be identical

Reply:
You're right, thank you for the suggestion, I've updated the code :)